vividus-plugin-web-app/src/test/java/org/vividus/proxy/steps/ProxyStepsTests.java

```diff
@@ -137,35 +145,89 @@ void checkHarEntryExistenceWithHttpMethodAndUrlPatternNoCalls() throws IOExcepti
         HarRequest harRequest = mock(HarRequest.class);
         when(harEntry.getRequest()).thenReturn(harRequest);
-        String key1 = "key1";
-        String value1 = "value1";
-        String key2 = "key2";
-        String value2 = "value2";
         HarQueryParam firstPair = new HarQueryParam();
-        firstPair.setName(key1);
-        firstPair.setValue(value1);
+        firstPair.setName(KEY1);
+        firstPair.setValue(VALUE1);
         HarQueryParam secondPair = new HarQueryParam();
-        secondPair.setName(key2);
-        secondPair.setValue(value2);
+        secondPair.setName(KEY2);
+        secondPair.setValue(VALUE2);
         when(harRequest.getQueryString())
                 .thenReturn(List.of(firstPair, secondPair));
         Set<VariableScope> variableScopes = Set.of(VariableScope.SCENARIO);
-        spy.captureRequestAndSaveURLQuery(httpMethod, URL_PATTERN, variableScopes, VARIABLE_NAME);
+        spy.captureRequestAndSaveURLQuery(HTTP_METHOD, URL_PATTERN, variableScopes, VARIABLE_NAME);
         verify(bddVariableContext).putVariable(eq(variableScopes), eq(VARIABLE_NAME), argThat(value ->
         {
             Map<String, String> map = (Map<String, String>) value;
             return map.size() == 2
-                    && map.containsKey(key1)
-                    && value1.equals(map.get(key1))
-                    && map.containsKey(key2)
-                    && value2.equals(map.get(key2));
+                    && map.containsKey(KEY1)
+                    && VALUE1.equals(map.get(KEY1))
+                    && map.containsKey(KEY2)
+                    && VALUE2.equals(map.get(KEY2));
+        }));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void checkCaptureRequestDataFromHarEntry() throws IOException
+    {
+        ProxySteps spy = spy(proxySteps);
+        HarEntry harEntry = mock(HarEntry.class);
+        doReturn(List.of(harEntry)).when(spy).checkNumberOfRequests(HTTP_METHOD, URL_PATTERN,
+                ComparisonRule.EQUAL_TO, 1);
+        HarResponse harResponse = mock(HarResponse.class);
+        when(harEntry.getResponse()).thenReturn(harResponse);
+        HarRequest harRequest = mock(HarRequest.class);
+        when(harEntry.getRequest()).thenReturn(harRequest);
+        HarQueryParam firstPair = new HarQueryParam();
+        firstPair.setName(KEY1);
+        firstPair.setValue(VALUE1);
+        HarQueryParam secondPair = new HarQueryParam();
+        secondPair.setName(KEY2);
+        secondPair.setValue(VALUE2);
+        when(harRequest.getQueryString())
+                .thenReturn(List.of(firstPair, secondPair));
+        String mimeType = "mimeType";
+        String text = "text";
+        String comment = "comment";
+        HarPostData postData = new HarPostData();
+        postData.setMimeType(mimeType);
+        postData.setText(text);
+        postData.setComment(comment);
+        HarPostDataParam postDataParam1 = new HarPostDataParam();
```

Comment:
constructions of HarPostDataParam, HarQueryParam objects can be moved into separate method, please request similar logic